### PR TITLE
Facebook API

### PR DESCRIPTION
**Pull request by vshtuka**

### CODE REVIEW

File: app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt [hunk 1]
```diff
@@ -1,46 +1,75 @@
...
-    fun getMosaicsList(): LiveData<List<Drawable>> {
-        return mosaicsList
+    private fun loadImagesFromInstagram(application: Application) {
+        val token: AccessToken = AccessToken.getCurrentAccessToken()
+        GraphRequest(
+            token,
+            "/$INSTAGRAM_ID/media",
+            null,
+            HttpMethod.GET
+        ) { response ->
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt]: Please put this callback into a new method

File: app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt [hunk 2]
```diff
@@ -1,46 +1,75 @@
...
+            val graphResponse: JSONObject = response.jsonObject
+            val data: JSONArray = graphResponse.getJSONArray("data")
+            for (id in 0 until data.length()) {
+                val imageId: String = data.getJSONObject(id).getString("id")
+                GraphRequest(
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt]: How can I cancel this operation? For example user minimized app or move to next screen.
Also please cover "No internet connection" or "Network problems" cases

**Review by FAQEnD — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/superimagesviewer/MosaicFragment.kt [hunk 3]
```diff
@@ -32,7 +32,12 @@ class MosaicFragment : Fragment() {
         mosaicViewModel = ViewModelProvider(this).get(MosaicViewModel::class.java)
         initRecyclerView()
     }
-    
+
+    override fun onStop() {
+        super.onStop()
+        MosaicRepository.graphRequest.executeAsync().cancel(true)
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/MosaicFragment.kt]: I think you did wrong way here. 
You want to have next result:
If application onStop() was called -> you cancel current request to graphAPI (findImagesByIds)
But what is really going on here is:
If application onStop() was called -> you start new async request with _executeAsync()_ method and in the same moment cancel it.
But old one still running..

Comment (FAQEnD) [reply app/src/main/java/com/example/superimagesviewer/MosaicFragment.kt]: In order to fix this you need to save into variable result of graphRequest.executeAsync()
I mean something like this:
var findImagesByIdsAsyncTask = graphRequest.executeAsync()

So then you will be able do like this:

onStop() {
MosaicRepo.findImagesByIdsAsyncTask.Cancel()
}

File: app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt [hunk 4]
```diff
@@ -31,25 +44,34 @@ class MosaicRepository(application: Application) {
...
+                HttpMethod.GET
+            ) { response2 ->
+                loadImageByCoil(response2, application)
+            }
+            graphRequest.executeAsync()
+        }
+    }
+
+    private fun loadImageByCoil(response2: GraphResponse, application: Application) {
+        val graphResponse2: JSONObject = response2.jsonObject
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt]: Please rename all variables like graphResponse2, response2, response, etc. to something more readable, like "findImagesByIdsRequest", "findImagesByIdsResponse", etc.

**Review by FAQEnD — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt (hunk 1 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt]: done

File: app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt (hunk 2 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt]: done

File: app/src/main/java/com/example/superimagesviewer/MosaicFragment.kt (hunk 3 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/MosaicFragment.kt]: done

File: app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt (hunk 4 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt]: done

**Review by FAQEnD — APPROVED** [review]

**Pull request merged**